### DISCUSSION TITLE
Popover mysteriously scrolls up on content interaction

### DISCUSSION THREAD

**Discussion by negue** [huntabyte/shadcn-svelte#1045] [Help] [answered]
I don't know if this is a Bug or if I'm missing something.

## The Issue
I have a Popover which holds a Json Viewer Component (source included in example), if I scrolled in the Popover down, and when I press on any of the brackets (to collapse them) it scrolls back up. 

Even though there is no "scrollTo" call or what ever in the Json Viewer component, I can't really figure out what is causing this scroll up (or scrollTop = 0, or what is called somewhere 😅 ) 

Is it wrong that I use overflow: auto? or something else I should've activated for that?

I thought it could be some "the content children size changed so I scroll up"-bug, but with a smaller example which changes the div height in the content, it didn't scrolled back up again

## Reproduce
1. Open https://stackblitz.com/edit/github-fcxn7g?file=src%2Froutes%2F%2Bpage.svelte
2. Press "Trigger Popover"
3. Scroll Inside the Popover Content down 
4. Press on any of the of `{` Brackets of the subEntry Object
5. It scrolls up

## System Info just in Case
My Project/Machine:
```

  System:
    OS: macOS 14.4.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 122.16 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.17.0/bin/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v18.17.0/bin/npm
    pnpm: 8.7.6 - ~/Library/pnpm/pnpm
  Browsers:
    Safari: 17.4.1
  npmPackages:
    bits-ui: ^0.21.3 => 0.21.3 
    lucide-svelte: ^0.368.0 => 0.368.0 
    svelte: 5.0.0-next.112 => 5.0.0-next.112 

```

But the same error is in the linked stackblitz example,
Tested it with Safari, Firefox, and Vivaldi (Chrome Engine) so I don't think its any Browser Engine specific issue.

Edit 1: Weirdly using the same content in a hover-card it doesn't scroll up again, since both are using Floating UI internally this "should be" possible to fix in popover as well, but no Idea yet what the exact difference is 😬

**Comment by negue** [comment, marked as answer]
I found the reason ( I think )

TLDR its: `disableFocusTrap={true}` that solved my issue.

I tried to reproduce the scrolling by adding a simple button that would collapse each node, using my additional button the "scrolling" up didn't happen, this was odd, since it happened on clicking the brackets. I don't know what brought me to the idea that could've been the focustrap, but changing that it really helped. 🎉 

So I can only think its because of the following:

- each node has the brackets, when pressing it it changes to a collapsed state  <<< thats the issue
- when a click happens
   - it focuses the node, 
   - click is handled => the previously focused node is gone/replaced by collapsed one, 
   - that means the element is blur'ed, and there the focusTrap kicks in (I assume) with that it selects the first element that be selected, which is at the top => scroll happens 😅

**Comment by mskierski** [comment]
I had similar issue using "react-boostrap" OverlayTrigger commponent. This component has a "trigger" attribute. When set to "focus" the popover shows up and then sets focus to the first element on the page (scrolling to the top of the page).

I changed this by replacing "focus" with "hover"
